test-case/check-pause-release-suspend-resume.sh

```diff
@@ -103,82 +106,107 @@ case $test_mode in
     *)
         die "Invalid test mode: $test_mode. Accepted test mode: playback; capture"
     ;;
-esac
-
-[[ -z $file_name ]] && file_name=$dummy_file
-
-
-logger_disabled || func_lib_start_log_collect
-
-setup_kernel_check_point
-
-dlogi "Entering audio stream expect script with: $cmd -D $pcm -r $rate -c $channel -f $fmt -vv -i $dummy_file -q"
-dlogi "Will enter suspend-resume cycle during paused period of audio stream process"
-
-rm -rf /tmp/sof-test.lock
-
-# expect is tcl language script
-#   catch: Evaluate script and trap exceptional returns
-#   after ms: Ms must be an integer giving a time in milliseconds.
-#       The command sleeps for ms milliseconds and then returns.
-expect <<AUDIO
-spawn $cmd -D $pcm -r $rate -c $channel -f $fmt -vv -i $dummy_file -q
-set i 1
-set sleep_t $sleep_period
-expect {
-    "#*+*\%" {
-        #audio stream (aplay or arecord) is active now and playing
-        puts "\r===== (\$i/$repeat_count) pb_pbm: Pause $cmd, then wait for ===== "
-        puts "\r(\$i/$repeat_count) pb_pbm: $sleep_t ms after pause"
-        send " "
-        after \$sleep_t
-        puts "Finished sleep. Confirming $cmd is paused."
-
-        #check audio stream status --- R == paused, else == not paused
-        set retval [catch { exec ps -C $cmd -o state --no-header } state]
-        puts "$cmd state = \$state"
-
-        if {[string equal \$state "R"] != 0} {
-            puts "$cmd is paused, will now enter suspend-resume cycle"
-        } else {
-            puts "$cmd is not paused. Exiting test and failing."
-            exit 1
-        }
+    esac
 
-        #enter suspend-resume cycle once per pause instance
-        set retval [catch { exec bash check-suspend-resume.sh -l 1 } msg]
-
-        #prints logs from suspend-resume test
-        puts \$msg
-        puts "Finished suspend-resume test"
-        if { \$retval } {
-            puts "suspend resume cycle has failed."
-            set error_code [lindex \$::errorCode {2}]
-            puts "errorCode was: \$error_code"
-            exit 1
-        }
+    # only run suspend/resume once for each loop.
+    # Use system default value if no sleep type is specified
+    if [ -n "${OPT_VAL['T']}" ]; then
+        sleep_opts="-l 1 -T ${OPT_VAL['T']}"
+    else
+        sleep_opts="-l 1"
+    fi
 
-        #sucessful suspend/resume cycle, now release audio stream
-        puts "\r(\$i/$repeat_count) pb_pbm: Release $cmd, then wait for"
-        puts "\r(\$i/$repeat_count) pb_pbm: \$sleep_t ms after resume"
-        send " "
-        after \$sleep_t
-
-        puts "Finished sleep after resume"
-        incr i
-        if { \$i > $repeat_count } { exit 0 }
-        exp_continue
-    }
+    [[ -z $file_name ]] && file_name=$dummy_file
 }
+
+run_test()
+{
+    for idx in $(seq 0 $((PIPELINE_COUNT - 1)))
```

Comment:
You can save one indentation level by moving the `for` loop to the `main()` function.

Reply:
Good suggestion. PR is updated.